### DISCUSSION TITLE
Mangal 3

### DISCUSSION THREAD

**Discussion by metafates** [metafates/mangal#61] [Announcements]
Just wanted to say that this project is *not abandoned!* Right now I'm working on **Mangal 3**, which will be the biggest update. It will introduce a **lua** based scrapers with rich library (headless chrome, http server and other stuff) and zero runtime dependencies. Since this fundamentally changes the way mangal works internally I'm rewriting everything from scratch.

At this point lua part is working already, but there's still a lot to do. You can monitor progress at [3.0.0 branch](https://github.com/metafates/mangal/tree/3.0.0)

**Comment by sentinm** [comment]
Any **previews**?

**Reply by metafates** [reply]
Sure, here's an usage example of "mini" mode (it uses prompts instead of full TUI, evolved version of an old "inline" mode). Sources with 🌙 are lua based, 🐹 are predefined ones written in Go (they are a bit faster and come by default). This example uses [this lua source](https://github.com/metafates/mangal/blob/3.0.0/examples/Manganelo.lua)

https://user-images.githubusercontent.com/62389790/182786984-7ec0a6c0-2cf3-4415-9c22-85b2d68de5e9.mov

**Comment by metafates** [comment]
Quick update! Everything is almost ready 🎉. The only things left are better styles for TUI and some polishing. Also I'd like to mention that codebase is much cleaner and more sustainable. Stay tuned!

https://user-images.githubusercontent.com/62389790/183158432-a2bcec6a-6a7e-4a8c-9635-4550f4bfa16f.mov

**Comment by metafates** [comment]
Mangadex now comes as a built-in provider

<img width="365" alt="Screenshot 2022-08-06 at 11 26 39" src="https://user-images.githubusercontent.com/62389790/183241202-450b6477-ac8a-4b66-914c-f693ebb92ac1.png">

**Comment by justchokingaround** [comment]
sheeeeesh

**Reply by metafates** [reply]
sheesh 🥶

**Reply by justchokingaround** [reply]
😎 😎 😎

**Comment by metafates** [comment]
Probably gonna release it today. But Anilist integration is not ready yet. Maybe in the next release... 🦥

**Reply by metafates** [reply]
nvm, I've just added it 😎
